src/authStrategies/RemoteAuth.js

```diff
@@ -0,0 +1,221 @@
+    }
+
+    async beforeBrowserInitialized() {
+        const puppeteerOpts = this.client.options.puppeteer;
+        const sessionDirName = this.clientId ? `session-${this.clientId}` : 'session';
+        const dirPath = path.join(this.dataPath, sessionDirName);
+
+        if (puppeteerOpts.userDataDir && puppeteerOpts.userDataDir !== dirPath) {
+            throw new Error('RemoteAuth is not compatible with a user-supplied userDataDir.');
+        }
+
+        this.userDataDir = dirPath;
+        this.sessionName = sessionDirName;
+
+        await this.extractRemoteSession();
+
+        this.client.options.puppeteer = {
+            ...puppeteerOpts,
+            userDataDir: dirPath
+        };
+    }
+
+    async logout() {
+        await this.deleteRemoteSession();
+
+        let pathExists = await this.isValidPath(this.userDataDir);
+        if (pathExists) {
+            await fs.promises.rm(this.userDataDir, {
+                recursive: true,
+                force: true
+            }).catch(() => {});
+        }
+        clearInterval(this.backupSync);
+    }
+
+    async authReady() {
+        const sessionExists = await this.store.sessionExists({session: this.sessionName});
+        if(!sessionExists) {
+            await this.delay(30000); /* Initial delay sync required for session to be stable enough to recover */
+            await this.storeRemoteSession();
+        }
+        var self = this;
+        this.backupSync = setInterval(async function () {
+            await self.storeRemoteSession();
+        }, this.backupSyncMs);
+    }
+
+    async storeRemoteSession() {
+        try {
+            const sessionExists = await this.store.sessionExists({session: this.sessionName});
+            if (sessionExists) {
+                await this.store.delete({session: this.sessionName});
+            }
+            /* Compress & Store Session */
+            const pathExists = await this.isValidPath(this.userDataDir);
+            if (pathExists) {
+                await this.compressSession();
+                await this.store.save({session: this.sessionName});
+                await fs.promises.unlink(`${this.sessionName}.zip`);
+            }
+        } catch (error) {
+            console.log('RemoteAuth Error => storeRemoteSession: ', error);
+        }
+    }
+
+    async extractRemoteSession() {
+        try {
+            const sessionExists = await this.store.sessionExists({session: this.sessionName});
+            const pathExists = await this.isValidPath(this.userDataDir);
+            if (pathExists) {
+                await fs.promises.rm(this.userDataDir, {
+                    recursive: true,
+                    force: true
+                });
+            }
+            if (sessionExists) {
+                await this.store.extract({session: this.sessionName});
+                await this.unCompressSession();
+                await fs.promises.unlink(`RemoteAuth-${this.sessionName}.zip`);
```

Comment:
Same comment here about being more explicit about file paths - currently this is making some assumptions as to where the files are and what they will be named

Reply:
I get the part of being more explicit with the file paths and I agree, but the naming assumbtion is a bit unclear to me.. you mean the `RemoteAuth-${this.sessionName}.zip` ? This name is being set the the zip file extracted on `unCompressSession()` of RemoteAuth class not on the store itself. Pls correct me if I'm missing something else

Reply:
> Done